hal/targets/hal/TARGET_Freescale/TARGET_KSDK2_MCUS/TARGET_K64F/storage_driver.c

```diff
@@ -463,17 +575,18 @@ static inline void setup8ByteWrite(uint64_t addr, const void *data)
 #endif /* ifdef USING_KSDK2 */
 }
 
+__RAMFUNC
 static inline void setupProgramSection(uint64_t addr, const void *data, size_t cnt)
 {
 #ifdef USING_KSDK2
     static const uintptr_t FlexRAMBase = FSL_FEATURE_FLASH_FLEX_RAM_START_ADDRESS;
-    memcpy((void *)FlexRAMBase, (const uint8_t *)data, cnt);
+    MEMCPY((void *)FlexRAMBase, (const uint8_t *)data, cnt);
```

Comment:
Taking this one step further. Why even go to all this trouble to try to persuade 3 different toolchains to get the code in RAM? (Which I'm not convinced we've totally ironed out yet).

Might it not be ultimately simpler to hand craft the RAM code, in a toolchain-independent way?

    // hand-assembled, untested
    // void setAndWaitForReadHighRamCode(uintptr_t reg, uint8_t bit)
    uint16_t setAndWaitForReadHighRamCode[] = {
        0x7802, //        LDRB    r2,[r0]
        0x430A, //        ORRS    r2,r2,r1
        0x7002, //        STRB    r2,[r0]
        0x7802, //        LDRB    r2,[r0]
        0x420A, //        TST     r2,r1
        0xD1FC, //        BEQ     2 instructions back
        0x4770, //        BX lr
    };

#define setAndWaitForReadHigh(reg, bit) ((void (*)(uintptr_t , uint8_t )) setAndWaitForReadHighRamCode)(reg, bit)

#define launchAndWaitForCommand() setAndWaitForReadHigh(FTFA_FSTAT, FTFA_FSTAT_CCIF_MASK)